extensions/machine-learning-services/src/modelManagement/registeredModelService.ts

```diff
@@ -56,22 +95,73 @@ export class RegisteredModelService {
 
 	private async runRegisteredModelsListQuery(connection: azdata.connection.ConnectionProfile): Promise<azdata.SimpleExecuteResult | undefined> {
 		try {
-			return await this._queryRunner.runQuery(connection, this.registeredModelsQuery(this._config.registeredModelDatabaseName, this._config.registeredModelTableName));
+			return await this._queryRunner.runQuery(connection, this.registeredModelsQuery(connection.databaseName, this._config.registeredModelDatabaseName, this._config.registeredModelTableName));
+		} catch {
+			return undefined;
+		}
+	}
+
+	private async runUpdateModelQuery(connection: azdata.connection.ConnectionProfile, model: RegisteredModel): Promise<azdata.SimpleExecuteResult | undefined> {
+		try {
+			return await this._queryRunner.runQuery(connection, this.getUpdateModelScript(connection.databaseName, this._config.registeredModelDatabaseName, this._config.registeredModelTableName, model));
 		} catch {
 			return undefined;
 		}
 	}
 
-	private registeredModelsQuery(databaseName: string, tableName: string) {
+	private registeredModelsQuery(currentDatabaseName: string, databaseName: string, tableName: string): string {
+		if (!currentDatabaseName) {
+			currentDatabaseName = 'master';
+		}
+		return `
+		${this.configureTable(databaseName, tableName)}
+		USE ${currentDatabaseName}
+		SELECT artifact_id, artifact_name, name, description, version, created from ${databaseName}.dbo.${tableName}
```

Comment:
>databaseName [](start = 80, length = 12)

The db and table names should be quoted and escaped.

And is dbo schema always guaranteed? Users can't use their own tables?

Reply:
right now this is a hacky way to make it work, I probably change this code all but for now need to have something working. 

---
In reply to: [385788932](https://github.com/microsoft/azuredatastudio/pull/9377#discussion_r385788932) [](ancestors = 385788932)